### PR TITLE
highway=motorway_link does not imply oneway, fixes #1662

### LINKED ISSUE #1662: motorway_link is not implied oneway
According to https://github.com/gravitystorm/openstreetmap-carto/issues/1820 and the wiki OSRM routing rules are incorrect.

### PR DESCRIPTION
**Pull request by daniel-j-h**
For #1662.

References:
- http://wiki.openstreetmap.org/wiki/Tag:highway%3Dmotorway_link#Tagging_oneway
- http://wiki.openstreetmap.org/wiki/Tag:highway%3Dmotorway_link
- http://wiki.openstreetmap.org/wiki/Tag:highway%3Dmotorway

### CODE REVIEW

File: features/guidance/motorway.feature [hunk 1]
```diff
@@ -176,11 +176,11 @@ Feature: Motorway Guidance
...
        When I route I should get
-            | waypoints | route           | turns                                |
-            | a,e       | abcde,abcde     | depart,arrive                        |
-            | f,e       | fgc,abcde,abcde | depart,merge slight left,arrive      |
-            | a,i       | abcde,chi,chi   | depart,off ramp slight right,arrive |
-            | f,i       | fgc,chi,chi     | depart,off ramp right,arrive        |
+            | waypoints | route           | turns                              |
+            | a,e       | abcde,abcde     | depart,arrive                      |
+            | f,e       | fgc,abcde,abcde | depart,merge slight left,arrive    |
+            | a,i       | abcde,chi,chi   | depart,on ramp slight right,arrive |
```
Comment (daniel-j-h) [inline features/guidance/motorway.feature]: a -> i should always have been on ramp and not off ramp, or am I missing something here?

File: features/guidance/motorway.feature [hunk 2]
```diff
@@ -194,11 +194,11 @@ Feature: Motorway Guidance
...
        When I route I should get
-            | waypoints | route           | turns                               |
-            | a,e       | abcde,abcde     | depart,arrive                       |
-            | f,e       | fgc,abcde,abcde | depart,merge slight right,arrive    |
-            | a,i       | abcde,chi,chi   | depart,off ramp slight left,arrive |
-            | f,i       | fgc,chi,chi     | depart,off ramp left,arrive        |
+            | waypoints | route           | turns                             |
+            | a,e       | abcde,abcde     | depart,arrive                     |
+            | f,e       | fgc,abcde,abcde | depart,merge slight right,arrive  |
+            | a,i       | abcde,chi,chi   | depart,on ramp slight left,arrive |
```
Comment (daniel-j-h) [inline features/guidance/motorway.feature]: same here /cc @MoKob can you clarify?

File: features/guidance/motorway.feature (hunk 1 above)
Comment (MoKob) [inline features/guidance/motorway.feature]: You are exiting the motorway abcde at this point. So chi is an off ramp (its not about getting on / off the ramp but whether you are leaving / entering a motorway)

File: features/guidance/motorway.feature (hunk 2 above)
Comment (MoKob) [inline features/guidance/motorway.feature]: its just a mirror of the scenario above, so same comment applies.

**Comment by daniel-j-h** [comment]
Fixed the on/off ramp test cases relying on implied oneway.

**Comment by TheMarex** [comment]
This will probably degrade the routing performance initially.

@planemad keep a look out to routing bugs that would occur here after the next OSRM release. Do you think we could add the non-tagged `motorway_link` roads to to-fix?

**Comment by planemad** [comment]
@TheMarex unless this causes routing failures, it might be hard to keep a track of routing bugs unless explicitly reported. In the meanwhile, we can detect such ways and load into to-fix for a community review.

**Pull request merged**